lib/ardb/has_slug.rb

```diff
@@ -53,11 +53,12 @@ def reset_slug
 
       def ardb_has_slug_generate_slug
         slug_attr_value = self.send(self.class.ardb_has_slug_config[:attribute])
-        if !slug_attr_value || slug_attr_value.to_s.empty?
-          slug_attr_value = self.instance_eval(&self.class.ardb_has_slug_config[:source_proc])
+        slug_source     = slug_attr_value
+        if !slug_source || slug_source.to_s.empty?
```

Comment:
@jcredding (sorry for the comment spam) Ahh, I think I might be understanding it now.  It works as is (by chance) but the code doesn't communicate intent very well.  I'd think you'd want to continue to use `slug_attr_value` for this if statement.  That is the intent is to set the source to the result of the source proc if there is not attr value, right?  Changing to `slug_attr_value` here would better communicate this I think.  This implementation confused the heck outta me.

I think this is all saying to me "switch back to something more like the orig implementation (which did all this in one if/else statement.  Maybe use the local var named `a` trick to make the code style nicely now that the attribute name is variable (which is a difference from the original implementation).

Also, I'm ok with the local var `slug_source` now that I better understand the intent.